sway/input/cursor.c

```diff
@@ -37,160 +37,86 @@ static uint32_t get_current_time_msec(void) {
 	return now.tv_sec * 1000 + now.tv_nsec / 1000000;
 }
 
-static struct wlr_surface *layer_surface_at(struct sway_output *output,
-		struct wl_list *layer, double ox, double oy, double *sx, double *sy) {
-	struct sway_layer_surface *sway_layer;
-	wl_list_for_each_reverse(sway_layer, layer, link) {
-		double _sx = ox - sway_layer->geo.x;
-		double _sy = oy - sway_layer->geo.y;
-		struct wlr_surface *sub = wlr_layer_surface_v1_surface_at(
-			sway_layer->layer_surface, _sx, _sy, sx, sy);
-		if (sub) {
-			return sub;
-		}
-	}
-	return NULL;
-}
-
-static bool surface_is_xdg_popup(struct wlr_surface *surface) {
-	struct wlr_xdg_surface *xdg_surface =
-		wlr_xdg_surface_try_from_wlr_surface(surface);
-	return xdg_surface != NULL && xdg_surface->role == WLR_XDG_SURFACE_ROLE_POPUP &&
-		xdg_surface->popup != NULL;
-}
-
-static struct wlr_surface *layer_surface_popup_at(struct sway_output *output,
-		struct wl_list *layer, double ox, double oy, double *sx, double *sy) {
-	struct sway_layer_surface *sway_layer;
-	wl_list_for_each_reverse(sway_layer, layer, link) {
-		double _sx = ox - sway_layer->geo.x;
-		double _sy = oy - sway_layer->geo.y;
-		struct wlr_surface *sub = wlr_layer_surface_v1_surface_at(
-			sway_layer->layer_surface, _sx, _sy, sx, sy);
-		if (sub && surface_is_xdg_popup(sub)) {
-			return sub;
-		}
-	}
-	return NULL;
-}
-
 /**
  * Returns the node at the cursor's position. If there is a surface at that
  * location, it is stored in **surface (it may not be a view).
  */
 struct sway_node *node_at_coords(
 		struct sway_seat *seat, double lx, double ly,
-		struct wlr_surface **surface, double *sx, double *sy) {
-	// find the output the cursor is on
-	struct wlr_output *wlr_output = wlr_output_layout_output_at(
-			root->output_layout, lx, ly);
-	if (wlr_output == NULL) {
-		return NULL;
-	}
-	struct sway_output *output = wlr_output->data;
-	if (!output || !output->enabled) {
-		// output is being destroyed or is being enabled
-		return NULL;
-	}
-	double ox = lx, oy = ly;
-	wlr_output_layout_output_coords(root->output_layout, wlr_output, &ox, &oy);
+		struct wlr_surface **_surface, double *sx, double *sy) {
```

Comment:
_surface, is actually used here, along with every other function paramater. The name is kept because that's what upstream has. As for the rename, again, this is what upstream chose to go with, and it makes sense: The function returns a `sway_node`; it giving you a `wlr_surface` is just a bonus.

Reply:
What do you mean by upstream in this case?

Reply:
Ah, I don't know how the underscore was added, I've restored this name to what upstream has: just `surface`